charts/cromwell/templates/metricsService.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: cromwell-metrics
```

Comment:
With a Service and ServiceMonitor for all pods, is it possible to differentiate between metrics for the different deployments? Eg. cromwell1-reader vs cromwell1-runner vs cromwell2?

Having a single Service collect for all pods in multiple Deployments _feels_ a little surprising/unusual to me, but it very well might be a Standard Prometheus Pattern. I definitely don't think it'll break anything!

Would it make sense to use PodMonitors instead, so we don't have to create a new Service?

https://github.com/prometheus-operator/prometheus-operator/issues/3119#issuecomment-610445145

Reply:
I did it this way because the service monitor is the standard pattern for prometheus. It can definitely still distinguish between all the different pods and cromwell service type. [Example](https://prometheus.dsde-dev.broadinstitute.org/graph?g0.range_input=1h&g0.expr=jvm_threads_current&g0.tab=0). 

We could use pod monitors if you feel strongly about it. That would require updating some configs on the prometheus side. How prometheus handles pod monitors and serviceMonitors are configured separately.

Pod monitor imo is more intended for when you want to correlate metrics across multiple systems. Something like appending a label with a trace id to metrics from a service and a database if both were running in kubernetes.

Reply:
IMO the service monitor pattern is particularly well suited for Cromwell because it enables us to easily group metrics from a crowell deployment as a whole while retaining the ability to drill down to the individual cromwell components. With pod monitor we would lose that ability to easily correlate metrics across a whole cromwell deployment as a each pod would be considered a separate entity.

Reply:
Nice! It looks like prometheus has included the pod name as a dimension no those metrics. Would it be possible to include the deployment type as a dimension as well?

Like, if I wanted to have a dashboard with just cromwell1 runner JVM metrics and exclude metrics for the other pods, could I?

Reply:
yes we the dashboard I made supports this. Probably easier to just show you if you are interested.